src/main/java/org/colonelkai/publictransit/line/LineDirection.java

```diff
@@ -0,0 +1,45 @@
+package org.colonelkai.publictransit.line;
+
+import org.colonelkai.publictransit.node.Node;
+
+import java.util.List;
+import java.util.Optional;
+
+public enum LineDirection {
+
+    //one way normal
+    POSITIVE_SINGLE(false, 1),
+    //two way normal
+    POSITIVE_FLIP(true, 1),
+    //one way reversed
+    NEGATIVE_SINGLE(false, -1);
+
+    private final int addition;
+    private final boolean canFlipBack;
+
+    LineDirection(boolean canFlipBack, int addition) {
+        this.addition = addition;
+        this.canFlipBack = canFlipBack;
+    }
+
+    public Optional<Node> getNextNode(int currentIndex, List<Node> nodes) {
```

Comment:
When it's both ways, or "positive flip" as you called it (I'm not sure what it means), the player could go positive 1 or negative 1 index through the node system, is this handled? Sorry if stupid quesiton.

Reply:
Thanks!!!